File: 1616-minimum-difference-between-largest-and-smallest-value-in-three-moves/minimum-difference-between-largest-and-smallest-value-in-three-moves.py
class Solution:
    def minDifference(self, nums: List[int]) -> int:
        nums.sort()
        n = len(nums)

        if len(nums) <= 4:
            return 0

        ans = min(nums[-4] - nums[0], nums[-1] - nums[3], nums[-3] - nums[1], nums[-2] - nums[2])

        return ans

File: 1616-minimum-difference-between-largest-and-smallest-value-in-three-moves/minimum-difference-between-largest-and-smallest-value-in-three-moves.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var minDifference = function(nums) {
    let n = nums.length;
    nums.sort((a, b) => a - b);

    if (n <= 4 ){
        return 0;
    }

    let ans = Math.min(nums[n - 1] - nums[3], nums[n - 2] - nums[2], nums[n - 3] - nums[1], nums[n - 4] - nums[0]);

    return ans;
};